specs/ChainSpec/NetworkSpec.md

```diff
@@ -0,0 +1,119 @@
+- [Announce account](#Announce-account)
+
+### Routing
+
+When a node needs to send a message to another peer, it checks in the routing table if it is connected to that peer, possibly not directed but through several hops. Then it select one of the shortest path to the target peer and send a [`RoutedMessage`](Messages.md#RoutedMessage) to the first peer in the path.
+
+When it receives a `RoutedMessage`, it check if it is the target, in that case consume the body of the message, otherwise it finds a route to the target following described approach and send the message again. Is is important that before routing a message each peer check signature from original author of the message, passing a message with invalid signature can result in ban for sender. It is not required however checking the content of the message itself.
+
+Each `RoutedMessage` is equipped with a time-to-live integer. If this message is not for us, we decrement by one this field, if value is 0, we drop this message, otherwise we forward it.
+
+#### Routing back
+
+It is possible that node A is known to B but not the other way around. In case node A sends a request that requires a response to B, the response is routed back through the same path used to send the message from A to B. This are special types of messages.
+
+- **Store Previous Link** Request Messages are those that expect a response, and a route back is stored. This will expire after some fixed timeout or after a response is delivered. We store hash of the message to identify sender.
+- **Get Previous Link** Find previous link for a message. Proposal: nodes can try to route the message if it doesn't have a direct connection to previous link anymore.
+
+### Synchronization
+
+Routing tables is kept up to date between the network with two mechanisms:
+
+- First sync: When two nodes connect with each other they exchange all information in their routing tables. More details in [Handshake](#Handshake).
+- Broadcasting: When a node learns new information that is added to the routing table it broadcast that information to all active node currently connected to it. More details in [Edge](#Edge) and [Announce Account](#Announce-account).
+
+## Security
+
+Messages exchanged between routed (both direct or routed) are cryptographically signed with sender private key. Nodes ID contains public key of each node that allow other peer to verify this messages. To keep secure communication most of this message requires some nonce/timestamp that forbid a malicious actor reuse a signed message out of context.
+
+In the case of routing messages each intermediate hop should verify that message hash and signatures are valid before routing to next hop.
+
+### Abusive behavior
+
+When a node A sends more than `MAX_PEER_MSG_PER_MIN` messages per minute to node B, it will be banned and unable to keep sending messages to it. This a protection mechanism against abusive node to avoid being spammed by some peers.
```

Comment:
This should be phrased as a recommendation, since it is not a requirement for the node to be protocol compliant, and other implementations might choose to use different protection mechanisms.

Reply:
Yes, but how it is implemented might affect other nodes in the network. Even if it is not mandatory, it is desired to know rules that will make my node being banned, so that I avoid violating them.